fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt

```diff
@@ -274,7 +274,7 @@ class DefaultTabsTrayController(
 
         tab?.let {
             val isLastTab = browserStore.state.getNormalOrPrivateTabs(it.content.private).size == 1
-            if (!isLastTab) {
+            if (!isLastTab || !browserStore.state.selectedTabId.equals(tabId)) {
```

Comment:
this looks fairly straightforward. Could you please add some unit tests to increase our confidence around the fix

Reply:
sure I'll add some unit tests